spec/factories/trainees.rb

```diff
@@ -316,6 +316,10 @@
       apply_application
     end
 
+    trait :with_invalid_apply_application do
+      association :apply_application, invalid_data: { "degrees" => { "XXLbvaRY42wP52hWiP78r94m" => { institution: "University of Warwick" } } }.to_json
```

Comment:
Couldn't find a way interpolate the slug, i.e same one as factory, so we've hardcoded but not ideal

Reply:
This should be https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#callbacks the more ideal way

Reply:
thanks, fixed